generic/MAPL_Generic.F90

```diff
@@ -4955,16 +4960,25 @@ recursive integer function MAPL_AddChildFromDSO(NAME, userRoutine, grid, ParentG
   call ESMF_VMGet(vm, mpiCommunicator=comm, __RC__)
   CHILD_META%t_profiler = TimeProfiler(trim(NAME), comm_world=comm)
 
+  ! Construct a valid shared object library name
+
+  ! 1. Get the basename of the file without the extension
+  shared_object_library_basename = get_file_basename(SharedObj)
+  ! 2. Get the system dso suffix
+  system_dso_suffix = get_system_dso_suffix()
+  ! 3. Make a new shared object name
+  shared_object_library_with_system_dso_suffix = trim(shared_object_library_basename)//trim(system_dso_suffix)
```

Comment:
```suggestion
  shared_object_library_with_system_dso_suffix = shared_object_library_basename // system_dso_suffix)
```
Trim not needed - we are carefully using deferred length strings

Reply:
@tclune No, it was necessary. I don't know why, but I needed them. It's the only way the no-suffix test worked.

Maybe that means I need trims in my StringUtils?

Reply:
Ahh. I have a thought. In the Ocean Grid Comp the `SharedObj` variable being passed in is:
```fortran
character(len=ESMF_MAXSTR)         :: sharedObj
```

Maybe that's the reason?

Reply:
Yes - StringUtils must be careful not to _generate_ padded strings.   But it should also trim any input strings.

Reply:
Okay. I think I have this now. Doing some testing.

Reply:
I've added trims to stringutils and removed these trims.